packages/samples/headless-react/src/components/numeric-facet/numeric-facet.class.tsx

```diff
@@ -0,0 +1,91 @@
+import {Component} from 'react';
+import {
+  buildNumericFacet,
+  NumericFacet as HeadlessNumericFacet,
+  NumericFacetState,
+  NumericFacetValue,
+  NumericRangeRequest,
+  Unsubscribe,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+interface BaseNumericFacetProps {
+  field: string;
+  facetId: string;
+  format: (n: number) => string;
+}
+
+interface AutomaticNumericFacetProps extends BaseNumericFacetProps {
+  generateAutomaticRanges: true;
+  currentValues?: NumericRangeRequest[];
+}
+
+interface ManualNumericFacetProps extends BaseNumericFacetProps {
+  generateAutomaticRanges: false;
+  currentValues: NumericRangeRequest[];
+}
+
+export class NumericFacet extends Component<
+  AutomaticNumericFacetProps | ManualNumericFacetProps
+> {
+  private controller: HeadlessNumericFacet;
+  public state: NumericFacetState;
+  private unsubscribe: Unsubscribe = () => {};
+
+  constructor(props: AutomaticNumericFacetProps | ManualNumericFacetProps) {
+    super(props);
+
+    this.controller = buildNumericFacet(engine, {
+      options: {
+        field: props.field,
+        facetId: props.facetId,
+        generateAutomaticRanges: props.generateAutomaticRanges,
+        ...(props.currentValues && {currentValues: props.currentValues}),
```

Comment:
Why not pass the props as they are? If `currentValues` is undefined, there will be an error log (I changed this to throw an Error in the upcoming DateFacet documentation PR).

Someone using headless should not need to carefully build the options object.

Reply:
Unfortunately, setting `currentValues` manually to `undefined` throws an error.

Reply:
Ah hmm, I _think_ I addressed this my PR. I will open it soon,